### PR TITLE
[receiver/httpcheck] refactor client length checking

### PR DESCRIPTION
**Pull request by tranngoclam**
**Description:** <Describe what has changed.>

- Fix typo (`clentErr`) in receiver/httpcheckreceiver.
- Simplify slice length checking (a nil slice also has length zero).

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: receiver/httpcheckreceiver/scraper.go [hunk 1]
```diff
@@ -46,12 +46,13 @@ func (h *httpcheckScraper) start(_ context.Context, host component.Host) (err er
 
 // scrape connects to the endpoint and produces metrics based on the response
 func (h *httpcheckScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
-	if h.clients == nil || len(h.clients) == 0 {
```
Comment (codeboten) [inline receiver/httpcheckreceiver/scraper.go:49]: Is there a case where this change catches something that didn't used to be caught? If `clients` is nil, it returns the error, if the length is 0, it returns the error.

**Review by codeboten — COMMENTED** [review]
thanks for the fixing the typo, please review & sign the CLA

File: receiver/httpcheckreceiver/scraper.go (hunk 1 above)
Comment (tranngoclam) [reply receiver/httpcheckreceiver/scraper.go:49]: No, it's the same logic. The new change is to remove redundant checking. If a slice is nil then its length is also zero

**Comment by tranngoclam** [comment]
@codeboten can you review the PR again? Thanks

**Pull request closed without merging**